orchagent/routeorch.cpp

```diff
@@ -833,6 +838,7 @@ void RouteOrch::doTask(Consumer& consumer)
                 else if (m_syncdRoutes.find(vrf_id) == m_syncdRoutes.end() ||
                     m_syncdRoutes.at(vrf_id).find(ip_prefix) == m_syncdRoutes.at(vrf_id).end() ||
                     m_syncdRoutes.at(vrf_id).at(ip_prefix) != RouteNhg(nhg, ctx.nhg_index) ||
+                    gRouteBulker.bulk_entry_pending_removal(route_entry) ||
```

Comment:
>bulk_entry_pending_removal [](http://example.com/codeflow?start=33&length=26)

This is a valid fix.

However I am wondering how you find this issue. By design, m_toSync will already merge the sequence SET1,DEL,SET2 to DEL,SET2, so I think your first issue is not existing in orchagent process.
ref: [https://github.com/Azure/sonic-swss/blob/a4c80c3d8713d32b73d9428683b8670b46d65fcd/orchagent/orch.cpp#L108](https://github.com/Azure/sonic-swss/blob/a4c80c3d8713d32b73d9428683b8670b46d65fcd/orchagent/orch.cpp#L108)

Is it true?

Reply:
I felt this is not a `SET1,DEL,SET2` problem. It sound more like a route was previously configured to a nexthop group. Afterward, there is a DEL request followed by a SET request to configure to the same nexthop group. Following the logic of route orch, it would first put the DEL into the bulk. Then for the SET, it will be ignored since the nexthop group is the same as it was before removal. So this is not a second SET operation involved but the SET operation is getting ignored.

Reply:
Hi @qiluo-msft
Thanks for reviewing this.
The issue does exist in orchagent.
I understood that logic but It takes effect only if all three operations are received together and then handled. 
When this issue occurred, the first SET is received and handled, and then the DEL and 2nd SET. In fact, it doesn’t matter how long between the first SET and the DEL. Only the fact that a routing entry with a same next hop existing matters.
Actually this is a similar scenario with issue #9434. The difference is that the next hops were different between the existing entry and the SET operation, but are same here.
In regression test, this resulted in router check failed.
Is that clear?